modifier.go

```diff
@@ -2,83 +2,100 @@ package main
 
 import (
 	"bufio"
+	"flag"
 	"fmt"
 	"log"
 	"os"
 	"regexp"
 	"strconv"
 	"strings"
+	"sync"
+	"time"
+
+	"github.com/markhaur/srt-time-modifier/config"
+	"github.com/pkg/errors"
 )
 
-func main() {
+var regexPattern = regexp.MustCompile("[0-9]{2}:[0-9]{2}:[0-9]{2},[0-9]{3} --> [0-9]{2}:[0-9]{2}:[0-9]{2},[0-9]{3}")
 
-	if len(os.Args) < 4 {
-		fmt.Println("Usage: go run modifier.go <inputfile> <outputfile> <modify value>")
-		return
+func main() {
+	var configPath string
+	var duration time.Duration
+	var inputFile string
+	var outputFile string
+	flag.StringVar(&configPath, "config", "./config.yml", "Path to configuration file.")
+	flag.DurationVar(&duration, "duration", 0, "Time modify duration. Must be negative for now.")
+	flag.StringVar(&inputFile, "i", "./input/Hz.Omer.S01E01.srt", "Path to input file.")
+	flag.StringVar(&outputFile, "o", "./output/Hz.Omer.S01E01.srt", "Path to output file.")
+	flag.Parse()
+
+	if inputFile != "" && outputFile != "" && duration < 0 {
+		err := process(inputFile, outputFile, duration)
+		if err != nil {
+			log.Fatalf("could not process file %s: %v", inputFile, err)
+		}
+		os.Exit(0)
 	}
 
-	inputFile := os.Args[1]
-	outputFile := os.Args[2]
-	modifyValue, _ := strconv.Atoi(os.Args[3])
+	configuration, err := config.GetConfiguration(configPath)
+	if err != nil {
+		log.Fatalf("could not load config from path %s: %v\n", configPath, err)
+	}
 
-	fmt.Printf("Processing %v\n", inputFile)
+	var wg sync.WaitGroup
+	for _, cf := range configuration.Files {
+		wg.Add(1)
+		go func(iFile string, oFile string, mVal time.Duration) {
+			defer wg.Done()
+			err := process(iFile, oFile, mVal)
+			if err != nil {
+				log.Printf("could not process file %s: %v\n", iFile, err)
+			}
+		}(cf.InputFile, cf.OutputFile, cf.ModifyValue)
+	}
+	wg.Wait()
+}
 
+func process(inputFile string, outputFile string, modifyValue time.Duration) error {
```

Comment:
i think we should rename here too. inputFile would've suitable for an os.File var. Since this is just a path, we should rename to:
```suggestion
func process(inputPath string, outputPath string, offset time.Duration) error {
```